packages/integrations/vercel/src/analytics.ts

```diff
@@ -0,0 +1,69 @@
+import { inject } from '@vercel/analytics';
+import type { Metric } from 'web-vitals';
+import { getCLS, getFCP, getFID, getLCP, getTTFB } from 'web-vitals';
+
+const vitalsUrl = 'https://vitals.vercel-analytics.com/v1/vitals';
+
+type Options = { path: string; analyticsId: string; debug?: boolean };
+
+const getConnectionSpeed = () => {
+	return 'connection' in navigator &&
+		navigator['connection'] &&
+		'effectiveType' in (navigator['connection'] as unknown as { effectiveType: string })
+		? (navigator['connection'] as unknown as { effectiveType: string })['effectiveType']
+		: '';
+};
+
+const sendToAnalytics = (metric: Metric, options: Options) => {
+	const body = {
+		dsn: options.analyticsId,
+		id: metric.id,
+		page: options.path,
+		href: location.href,
+		event_name: metric.name,
+		value: metric.value.toString(),
+		speed: getConnectionSpeed(),
+	};
+
+	if (options.debug) {
+		// eslint-disable-next-line no-console
+		console.log('[Analytics]', metric.name, JSON.stringify(body, null, 2));
+	}
+
+	const blob = new Blob([new URLSearchParams(body).toString()], {
+		// This content type is necessary for `sendBeacon`
+		type: 'application/x-www-form-urlencoded',
+	});
+	if (navigator.sendBeacon) {
+		navigator.sendBeacon(vitalsUrl, blob);
+	} else
+		fetch(vitalsUrl, {
+			body: blob,
+			method: 'POST',
+			credentials: 'omit',
+			keepalive: true,
+		});
+};
+
+function webVitals() {
+	const options = {
+		path: window.location.pathname,
+		analyticsId: (import.meta as any).env.PUBLIC_VERCEL_ANALYTICS_ID,
+	};
```

Comment:
Do you think it matters that we don’t check `PUBLIC_VERCEL_ANALYTICS_ID` is defined before using it? We could return early if we wanted.

```suggestion
	const analyticsId = (import.meta as any).env.PUBLIC_VERCEL_ANALYTICS_ID;
	if (!analyticsId) {
		console.error('[Analytics] VERCEL_ANALYTICS_ID not found');
		return;
	}
	const options = { path: window.location.pathname, analyticsId };
```

Reply:
Vercel will add this env automatically. Sure it'll be better if we check it earlier.